mesonbuild/compilers/cpp.py

```diff
@@ -1128,3 +1129,242 @@ def get_option_std_args(self, target: BuildTarget, env: Environment, subproject:
+            self._create_combo_option('tgt_fp', "Target floating point processing", self.floating_point),
+            self._create_combo_option('os_impl', "Basic operating system functions", self.os_impl),
+        ])
+        return opts
+
+    def _get_option_common_args(self, target: 'BuildTarget', env: 'Environment', subproject: T.Optional[str] = None) -> T.List[str]:
+        args: T.List[str] = []
+
+        if self.get_compileropt_value('lic_wait', env, target, subproject):
+            args.append('-Xlicense-wait')
+        if self.get_compileropt_value('subprog_cmd', env, target, subproject):
+            args.append('-#')
+
+        args.append('-t{0}{1}{2}:simple'.format(
+            self.get_compileropt_value('tgt_proc', env, target, subproject),
+            self.object_format[self.get_compileropt_value('tgt_fmt', env, target, subproject)],
+            self.floating_point[self.get_compileropt_value('tgt_fp', env, target, subproject)],
+        ))
+
+        return args
+
+    def get_option_compile_args(self, target: 'BuildTarget', env: 'Environment', subproject: T.Optional[str] = None) -> T.List[str]:
+        args = self._get_option_common_args(target, env, subproject)
+
+        if not self.get_compileropt_value('eh', env, target, subproject):
+            args += self._apply_prefix(self.Prefix.CXX, '-Xexceptions-off')
+        if not self.get_compileropt_value('rtti', env, target, subproject):
+            args += self._apply_prefix(self.Prefix.CXX, '-Xrtti-off')
+
+        return args
+
+    def get_option_std_args(self, target: BuildTarget, env: Environment, subproject: T.Optional[str] = None) -> T.List[str]:
+        std = self.get_compileropt_value('std', env, target, subproject)
+        assert isinstance(std, str)
+        if std == 'c++98':
+            return self._apply_prefix(self.Prefix.CXX, '-Xstrict-ansi')
+        else:
+            return []
+
+    def get_option_link_args(self, target: 'BuildTarget', env: 'Environment', subproject: T.Optional[str] = None) -> T.List[str]:
+        args = self._get_option_common_args(target, env, subproject)
+
+        if self.get_compileropt_value('map', env, target, subproject):
+            args += self.linker._apply_prefix(['-m2', '-Xunused-sections-list', '-Xcheck-overlapping', f'-@O={target.filename}.map'])
+
+        return args
+
+    def compute_parameters_with_absolute_paths(self, parameter_list: T.List[str], build_dir: str) -> T.List[str]:
+        for idx, i in enumerate(parameter_list):
+            if i[:2] == '-I' or i[:2] == '-L':
+                parameter_list[idx] = i[:2] + os.path.normpath(os.path.join(build_dir, i[2:]))
+
+        return parameter_list
+
+    def get_always_args(self) -> T.List[str]:
+        """Disable super's large-file-support"""
+        return []
+
+    def get_compiler_check_args(self, mode: CompileCheckMode) -> T.List[str]:
+        return super(CPPCompiler, self).get_compiler_check_args(mode)# + ['-fpermissive']
```

Comment:
This looks like it just does a super() call? an it be dropped.

Reply:
`CPPCompiler` adds the `-fpermissive` flag which Diab doesn't accept. My intention here is to skip that by directly returning `CPPCompiler`'s super()'s output.

The commented `# + ['-fpermissive']` should be removed.

Reply:
Okay.

Side note to self, we should not just blindly add `-fpermissive` (or equivalent)